### PR TITLE
added initial project

### PR DESCRIPTION
**Pull request by oliver-krauss**
Publication of initial project

### CODE REVIEW

**Review by pointan — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,10 +1,89 @@
...
+To load an extension, you need to copy the files to your VS Code extensions folder .vscode/extensions. 
+
+Depending on your platform, it is located in the following folders:
+* Windows %USERPROFILE%\.vscode\extensions
+* macOS ~/.vscode/extensions
+* Linux ~/.vscode/extensions
+
+
+### Activate
+The editor will be automatically starting, if the language fsh is identified
```
Comment (aschuler84) [inline README.md]: will be started automatically

File: README.md [hunk 2]
```diff
@@ -1,10 +1,89 @@
...
+
+### Features
+#### Codestyling 
+Different colorization for declaration keywords, additional keyword, comments, and todos.
```
Comment (aschuler84) [inline README.md]: coloring?

File: extension.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+}
+exports.activate = activate;
+
+// this method is called when your extension is deactivated
+function deactivate() {}
+
+module.exports = {
+	activate,
+	deactivate
+}
```
Comment (aschuler84) [inline extension.js]: Remove vs-code template commands

File: language_server/providers/completion.js [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+            label: 'insert',
+            kind: vscode_languageserver.CompletionItemKind.Text,
+            data: 33
+        },{
+            label: 'Reference',
+            kind: vscode_languageserver.CompletionItemKind.Text,
+            data: 34
+        }
+    ];
+}
```
Comment (aschuler84) [inline language_server/providers/completion.js:176]: make return value of method an attribute. Currently, the Array structure is created new in every call. Is this intentionally?

File: language_server/providers/completion.js [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+//this function will add some additional information for the item selectet in the completion list.
+const onCompletionResolve = (item) => {
+    if (item.data === 1) {
+        item.detail = "Parent details";
+        item.documentation = "Specifies the base class for a profile or extension.\n\n"+
+        "For more information look at the HL7 FHIR Shorthand language reference.";
+    }
+    else if (item.data === 1) {
+        item.detail = "Id details";
```
Comment (aschuler84) [inline language_server/providers/completion.js:186]: elseif and if check for the same value

File: language_server/providers/completion.js [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+        item.documentation = "An identifier for an item.\n\n"+
+        "For more information look at the HL7 FHIR Shorthand language reference.";
+    }
+    else if (item.data === 2) {
```
Comment (aschuler84) [inline language_server/providers/completion.js]: provide an enum/constants for the values being checked

File: package.json [hunk 7]
```diff
@@ -0,0 +1,89 @@
+{
+	"name": "fsh-editor",
+	"displayName": "fsh_editor",
+	"description": "An interactiv editor for designing and writing FHIR-Shorthand specifications",
```
Comment (aschuler84) [inline package.json:4]: why is there a difference - fsh-editor vs. fsh_editor

File: src/extension.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+"use strict";
+const vscode = require('vscode');
+
+//as soon as the require is called, the providers are registered
+const completionProvider = require("./providers/completion");
+
+//this definisions are for the language server
```
Comment (aschuler84) [inline src/extension.js]: definitions

File: src/extension.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+/**
+ * @param {vscode.ExtensionContext} context
+ */
+function activate(context) {
+	// This line of code will only be executed once when your extension is activated
+    console.log('Congratulations, your extension "fsh-editor" is now active!');
+
+	// The command has been defined in the package.json file
+	// The commandId parameter must match the command field in package.json
+	let disposable = vscode.commands.registerCommand('fsh-editor.helloWorld', () => {
```
Comment (aschuler84) [inline src/extension.js]: remove irrelevant/unused dummy code

File: src/extension.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+        // The code you place here will be executed every time your command is executed
+
+        // Display a message box to the user
+        vscode.window.showInformationMessage('Hello World from fsh_editor!')
```
Comment (aschuler84) [inline src/extension.js]: not sure if this is still needed

File: src/parser/rules_parser/parse.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    } else if ( parser.results.length == 1){
+        console.log("one possibilities");
+    }
+    else {
+        console.log("no parser found");
+    }
+    console.log("finished");
+} catch (err){
+    console.log("ERROR at character " + err.stack)
+}
```
Comment (aschuler84) [inline src/parser/rules_parser/parse.js]: Is logging to the console the desired way to handle parsing errors?

File: test/suite/index.js [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+				console.error(err);
+				e(err);
+			}
+		});
+	});
+}
+
+module.exports = {
+	run
+};
```
Comment (aschuler84) [inline test/suite/index.js:51]: Are there any tests?

**Review by aschuler84 — CHANGES_REQUESTED** [review]
Some minor changes. However, I skipped the Lexer/Parser files.

File: README.md (hunk 2 above)
Comment (oliver-krauss) [reply README.md]: Technically yes as colorization means to color in something that is usually b/w vs coloring meaning that something is presented in a color.

File: extension.js (hunk 3 above)
Comment (oliver-krauss) [reply extension.js]: Yeah, I just commited the code I didn't write it, so I won't touch any of your code comments. Your choice if you merge or assign @SophieBauernfeind

File: src/extension.js (hunk 8 above)
Comment (pointan) [reply src/extension.js]: And I guess it should be `these` because definitions is plural.

**Review by aschuler84 — APPROVED** [review]

**Pull request merged**